plugins/kubernetes/app/models/kubernetes/role_validator.rb

```diff
@@ -31,7 +31,7 @@ def validate
       validate_not_matching_team
       validate_stateful_set_service_consistent
       validate_stateful_set_restart_policy
-      validate_annotations || validate_prerequisites
+      validate_annotations || validate_prerequisites_kinds || validate_prerequisites_consistency
```

Comment:
the || hack is here since we cannot run prerequisites checks when annotations are broken, so `||` chaining them makes no sense

```suggestion
      if validate_annotations
        validate_prerequisites_kinds
        validate_prerequisites_consistency
      end
```